exec/java-exec/src/main/java/org/apache/drill/exec/coord/zk/ZookeeperClient.java

```diff
@@ -227,13 +230,33 @@ public void create(final String path) {
 
     final String target = PathUtils.join(root, path);
     try {
-      curator.create().withMode(mode).forPath(target);
+      curator.create().creatingParentsIfNeeded().withMode(mode).forPath(target);
       getCache().rebuildNode(target);
     } catch (final Exception e) {
       throw new DrillRuntimeException("unable to put ", e);
     }
   }
 
+  public void createAsTransaction(List<String> paths) {
+    Preconditions.checkNotNull(paths, "no paths provided to create");
```

Comment:
Is it also better to check for empty list of paths? Also it might be good to add a comment for this function.